db/orders.js

```diff
@@ -0,0 +1,169 @@
+const {client} = require('./client');
+const { getProductById } = require('./products');
+
+async function createOrder({ status, userId }) {
+    const { rows: [order] } = await client.query(`
+        INSERT INTO orders(status, "userId")
+        VALUES ($1, $2)
+        RETURNING *;
+    `, [status, userId]);
+
+    return order;
+};
+
+async function addProductsToOrder(order){
+    try {
+        order.product = await getProductById(order.productId);
+
+        delete order.productId;
+
+        return order;
+    } catch (error) {
+        throw error;
+    };
+};
+  
+async function getAllOrders(){
+    try {
+        const {rows : orders} = await client.query(`
+        SELECT * FROM orders
+        LEFT JOIN order_products on orders.id = order_products."orderId";
+    `);
+
+    const newOrders = await Promise.all(orders.map(addProductsToOrder));
```

Comment:
This will create a separate query for every single order.  What should happen is one query for one order, getting back all the products.